server/py/services/api/tests/unit/db/test_artifacts.py

```diff
@@ -2793,6 +2794,110 @@ def test_delete_parent_artifacts(self):
         artifacts = self._db.list_artifacts(session=self._db_session, project=project)
         assert artifacts == []
 
+    @pytest.mark.parametrize(
+        "case",
+        [
+            {"kind": "hint", "attach_tags": False},
+            {"kind": "hint", "attach_tags": True},
+            {"kind": "default-check", "ids": [], "with_entities": None, "attach_tags": False, "expected": True},
+            {"kind": "default-check", "ids": [], "with_entities": [], "attach_tags": False, "expected": True},
+            {"kind": "default-check", "ids": ["non-empty"], "with_entities": None, "attach_tags": False, "expected": False},
+            {"kind": "default-check", "ids": None, "with_entities": None, "attach_tags": True, "expected": True},
+        ],
+        ids=[
+            "mysql-hint-default-query",
+            "mysql-hint-with-attach-tags",
+            "empty-ids-is-default",
+            "empty-ids-and-with-entities-are-default",
+            "non-empty-ids-breaks-default",
+            "attach-tags-still-default",
+        ],
+    )
+    def test_default_query_mysql_hint_and_defaults_with_attach_tags(self, monkeypatch, case):
+        if case["kind"] == "hint":
+            artifact_key = "dummy-artifact-for-default-query"
+            self._db.store_artifact(
+                self._db_session,
+                key=artifact_key,
+                artifact=self._generate_artifact(artifact_key, project=self.project),
+                project=self.project,
+            )
+
+            hint_called_flag = {"value": False}
+
+            original_with_hint = Query.with_hint
+
+            def with_hint_spy(self_query_obj, selectable, text, dialect_name=None):
+                if dialect_name == "mysql" and "USE INDEX" in str(text):
+                    hint_called_flag["value"] = True
+                return original_with_hint(self_query_obj, selectable, text, dialect_name=dialect_name)
+
+            monkeypatch.setattr(Query, "with_hint", with_hint_spy, raising=True)
+
+            _ = self._db._find_artifacts(
+                self._db_session,
+                project=self.project,
+                ids=None,
```

Comment:
why setting all function parameters and not *just* the parameters we _consider_ as defaults for UI purposes? that way, changing on the function parameters will trigger a failure when calling find_artifacts
the purpose is that if we change ui defaults, e.g now we find artifacts with parent_uri - meaning we change the default function signature - we should see this UT test failing. is that the case here?